src/v/raft/tests/membership_test.cc

```diff
@@ -269,3 +271,44 @@ FIXTURE_TEST(replace_whole_group, raft_test_fixture) {
         BOOST_REQUIRE_EQUAL(new_leader.consensus->config().brokers().size(), 1);
     }
 }
+
+FIXTURE_TEST(
+  make_sure_group_is_writable_during_config_replace, raft_test_fixture) {
+    // raft group with single replica
+    raft_group gr = raft_group(raft::group_id(0), 1);
+    gr.enable_all();
+
+    info("replicating some batches");
+    auto res = replicate_random_batches(gr, 5).get();
+    gr.create_new_node(model::node_id(5));
+    auto broker = gr.get_member(model::node_id(5)).broker;
+    gr.disable_node(model::node_id(5));
+    std::vector<model::broker> new_members;
+    new_members.reserve(1);
+    new_members.push_back(broker);
+    // replace configuration with other node, the target node is stopped
+    // to keep the transient state in which the old node is the only voter in
+    // raft group
+    info("replacing configuration");
+    res = retry_with_leader(gr, 5, 5s, [new_members](raft_node& leader) {
+              return leader.consensus
+                ->replace_configuration(new_members, model::revision_id(0))
+                .then([](std::error_code ec) {
+                    info("configuration replace result: {}", ec.message());
+                    return !ec
+                           || ec
+                                == raft::errc::configuration_change_in_progress;
+                });
+          }).get();
+    // if we failed to update configuration do nothing
+    if (!res) {
+        return;
```

Comment:
It seems like we don't need to permit this -- inside the controlled environment of a unit test we shouldn't get a configuration_change_in_progress (or any other error)

Reply:
good point. changed